### PR TITLE
UI fixes

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 1]
```diff
@@ -45,7 +45,12 @@ T get_response(QDBusMessage response) {
   QVariant first =  response.arguments().at(0);
   QDBusVariant dbvFirst = first.value<QDBusVariant>();
   QVariant vFirst = dbvFirst.variant();
-  return vFirst.value<T>();
+  if(vFirst.canConvert<T>()){
+    return vFirst.value<T>();
+  }else{
+    qDebug()<<"VARIANT UNPACKING FAILURE!!!!!";
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifiManager.cc:51]: LOGE? And a more normal error message with less `!` please.

File: selfdrive/ui/qt/offroad/networking.cc [hunk 2]
```diff
@@ -319,14 +319,15 @@ void WifiUI::refresh() {
...
-        ssid = ssid.left(30)+"…";
+      if(ssid.length() > 20){
+        ssid = ssid.left(20 - 3) + "…";
       }
       QLabel *ssid_label = new QLabel(ssid);
       ssid_label->setStyleSheet(R"(
         font-size: 55px;
       )");
-      hlayout->addWidget(ssid_label);
+      ssid_label->setFixedWidth(700);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/networking.cc:329]: why add these fixed sizes everywhere? Qt handles this nicely if we give it the right constraints

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 3]
```diff
@@ -45,7 +45,12 @@ T get_response(QDBusMessage response) {
   QVariant first =  response.arguments().at(0);
   QDBusVariant dbvFirst = first.value<QDBusVariant>();
   QVariant vFirst = dbvFirst.variant();
-  return vFirst.value<T>();
+  if(vFirst.canConvert<T>()){
+    return vFirst.value<T>();
+  }else{
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/wifiManager.cc:50]: add whitespace

File: selfdrive/ui/qt/offroad/networking.cc [hunk 4]
```diff
@@ -336,6 +337,7 @@ void WifiUI::refresh() {
       icon->setPixmap(pix.scaledToWidth(100, Qt::SmoothTransformation));
       icon->setSizePolicy(QSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed));
       hlayout->addWidget(icon, 0, Qt::AlignRight);
+      // hlayout->addSpacing(150);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/networking.cc:340]: uncomment or remove the line

File: selfdrive/ui/qt/offroad/networking.cc [hunk 5]
```diff
@@ -44,9 +45,19 @@ std::string exec(const char* cmd) {
 // Networking functions
 
 Networking::Networking(QWidget* parent, bool show_advanced) : QWidget(parent){
-  try {
-    wifi = new WifiManager(this);
-  } catch (std::exception &e) {
+  bool connected = false;
+  for(int i = 0 ; i < 20 ; i++) {
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/networking.cc:49]: doesn't this delay startup? I'd just move the wifi init to another function and try to init on the refresh callback if we haven't yet

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 6]
```diff
@@ -229,10 +234,12 @@ void WifiManager::deactivate_connections(QString ssid) {
     QString active_connection = active_connection_raw.path();
     QDBusInterface nm(nm_service, active_connection, props_iface, bus);
     QDBusObjectPath pth = get_response<QDBusObjectPath>(nm.call("Get", connection_iface, "SpecificObject"));
-    QString Ssid = get_property(pth.path(), "Ssid");
-    if (Ssid == ssid) {
-      QDBusInterface nm2(nm_service, nm_path, nm_iface, bus);
-      nm2.call("DeactivateConnection", QVariant::fromValue(active_connection_raw));// TODO change to disconnect
+    if (pth.path()!="" && pth.path()!="/") {
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/wifiManager.cc:237]: add whitespace

File: selfdrive/ui/qt/offroad/networking.cc [hunk 7]
```diff
@@ -43,29 +44,39 @@ std::string exec(const char* cmd) {
...
+Networking::Networking(QWidget* parent, bool show_advanced) : QWidget(parent), show_advanced(show_advanced){
+  s = new QStackedLayout;
+
+  QLabel* warning = new QLabel("Network manager is inactive!");
+  warning->setStyleSheet(R"(font-size: 65px;)");
+
+  s->addWidget(warning);
+  setLayout(s);
+
+  startup_timer = new QTimer(this);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/networking.cc:56]: why do we need another timer? can't we just use the refresh timer?

File: selfdrive/ui/qt/offroad/networking.cc [hunk 8]
```diff
@@ -43,29 +44,39 @@ std::string exec(const char* cmd) {
...
 
   QVBoxLayout* vlayout = new QVBoxLayout;
 
   if (show_advanced) {
     QPushButton* advancedSettings = new QPushButton("Advanced");
     advancedSettings->setStyleSheet(R"(margin-right: 30px)");
-    advancedSettings->setFixedSize(300, 100);
-    connect(advancedSettings, &QPushButton::released, [=](){s->setCurrentIndex(1);});
+    advancedSettings->setFixedSize(350, 100);
+    connect(advancedSettings, &QPushButton::released, [=](){s->setCurrentIndex(2);});
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/networking.cc:79]: use setCurrentWidget for cases like this, it's much easier to read and robust to changes like this

File: selfdrive/ui/qt/offroad/networking.cc (hunk 7 above)
Comment (grekiki) [reply selfdrive/ui/qt/offroad/networking.cc:56]: Do you think that would make the code better? I can do it, no problem.

**Comment by grekiki** [comment]
Onroad could theoretically lag if network manager is never detected... Is this a problem?

File: selfdrive/ui/qt/offroad/networking.cc [hunk 9]
```diff
@@ -43,29 +44,39 @@ std::string exec(const char* cmd) {
...
   try {
     wifi = new WifiManager(this);
   } catch (std::exception &e) {
-    QLabel* warning = new QLabel("Network manager is inactive!");
-    warning->setStyleSheet(R"(font-size: 65px;)");
-
-    QVBoxLayout* warning_layout = new QVBoxLayout;
-    warning_layout->addWidget(warning, 0, Qt::AlignCenter);
-    setLayout(warning_layout);
+    delete(wifi);
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/networking.cc:67]: Don't think you need to delete here.

**Pull request merged**